### PR TITLE
File Backup on Quadratic Cloud

### PR DESCRIPTION
**Pull request by davidkircos**
Data is being successfully inserted
CleanShot 2023-04-14 at 13 22 53

### CODE REVIEW

File: quadratic-api/src/routes/files.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+const ai_rate_limiter = rateLimit({
+  windowMs: Number(process.env.RATE_LIMIT_AI_WINDOW_MS) || 3 * 60 * 60 * 1000, // 3 hours
+  max: Number(process.env.RATE_LIMIT_AI_REQUESTS_MAX) || 25, // Limit number of requests per windowMs
+  standardHeaders: true, // Return rate limit info in the `RateLimit-*` headers
+  legacyHeaders: false, // Disable the `X-RateLimit-*` headers
+  keyGenerator: (request: JWTRequest, response) => {
+    return request.auth?.sub || 'anonymous';
+  },
+});
```
Comment (jimniels) [inline quadratic-api/src/routes/files.ts:22]: Sounds like you want to increase this limit quite a bit more than what the AI was doing

Comment (jimniels) [reply quadratic-api/src/routes/files.ts]: Fixed here https://github.com/quadratichq/quadratic/commit/502547399aae81cd2d13b454e14d81c9924024bd

**Review by jimniels — APPROVED** [review]
This looks good to me, though I'm not a good person to review DB stuff ha.

Left one comment, feel free to address or ignore, otherwise 👍

File: quadratic-api/src/routes/files.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        name: file_metadata.name,
+        contents: file_contents,
+        created_date: new Date(file_metadata.created),
+        updated_date: new Date(file_metadata.modified),
+        version: file_metadata.version,
+      },
+    });
+  }
+
+  response.status(200).json({ message: 'File backup successful.' });
```
[the remark this answers is not in the document]
Comment (davidkircos) [reply quadratic-api/src/routes/files.ts:69]: I believe this is caught globally in the express def in index.ts

**Pull request merged**